engine/src/execution_graph/logic_controllers/BatchJoinProcessing.h

```diff
@@ -226,10 +226,16 @@ class PartwiseJoin : public kernel {
 													right_join_types.cbegin(), right_join_types.cend());
 	}
 
-	std::unique_ptr<ral::frame::BlazingTable> join_set(const ral::frame::BlazingTableView & table_left, const ral::frame::BlazingTableView & table_right){
+	std::unique_ptr<ral::frame::BlazingTable> join_set(
+		const ral::frame::BlazingTableView & table_left,
+		const ral::frame::BlazingTableView & table_right,
+		std::string condition = "")
+	{
 		std::unique_ptr<CudfTable> result_table;
 		std::vector<std::pair<cudf::size_type, cudf::size_type>> columns_in_common;
-		if(this->join_type == INNER_JOIN) {
+
+		// cross join  (condition=[true])
+		if (condition == "true") {
```

Comment:
Rather than using and passing around this string literal `condition` you should detect this in the construction of the join kernel and set the `this->join_type` variable to a new enum `CROSS_JOIN`

Reply:
ok!